src/inmanta/agent/agent.py

```diff
@@ -311,43 +311,49 @@ async def execute(
         self, dummy: "ResourceActionBase", generation: "Dict[ResourceIdStr, ResourceActionBase]", cache: AgentCache
     ) -> None:
         await dummy.future
-        try:
-            result = await self.scheduler.get_client().get_resource(
-                self.scheduler.agent._env_id,
-                self.resource_id.resource_version_str(),
-                logs=True,
-                log_action=const.ResourceAction.deploy,
-                log_limit=1,
-            )
-            if result.code != 200 or result.result is None:
-                LOGGER.error("Failed to get the status for remote resource %s (%s)", str(self.resource_id), result.result)
-                return
+        async with self.scheduler.agent.process.cad_ratelimiter:
+            try:
+                # got event or cancel first
+                if self.is_done():
+                    return
 
-            status = const.ResourceState[result.result["resource"]["status"]]
-            self.running = True
-            if status in const.TRANSIENT_STATES or self.future.done():
-                # wait for event
-                pass
-            else:
-                if "logs" in result.result and len(result.result["logs"]) > 0:
-                    log = result.result["logs"][0]
+                self.running = True
+                result = await self.scheduler.get_client().get_resource(
```

Comment:
I've got some data from production

I'm afraid we do a LOT of these, we may want to batch them up.

The ability to cancel them (we add here), may also help somewhat. 

```json
"rpc.get_resource": {
    "avg": 0.2436818535260758,
    "sum": 19656.353032827377,
    "count": 80664.0,
    "max": 3.0693390369415283,
    "min": 0.0059680938720703125,
    "std_dev": 0.13372318645433742,
    "15m_rate": 1.459812261442336,
    "5m_rate": 0.7380262996161672,
    "1m_rate": 0.007291537878714465,
    "mean_rate": 1.008132603019391,
    "50_percentile": 0.14774608612060547,
    "75_percentile": 0.19217169284820557,
    "95_percentile": 0.3021209955215453,
    "99_percentile": 0.48956055402755744,
    "999_percentile": 1.5138091018199922
  },
```